Paract.15_06/Task 2/CarColection.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+namespace Paract._15_06.Task_2
+{
+    internal class CarColection<T> where T : Car
+    {
+        public T[] carArr;
+
+        public CarColection()
+        {
+            this.carArr = new T[0];
+        }
+
+        T this[string name]
+        {
+
+            get
+            {
+                T car = null;
+                for (int i = 0; i < carArr.Length; i++)
+                {
+                    if (carArr[i].Name.Equals(name) == true)
```

Comment:
No need to compare bool with bool.
It is enough to leave 'carArr[i].Name.Equals(name)' only.